planemo/workflow_lint.py

```diff
@@ -60,21 +60,52 @@ class WorkflowLintContext(LintContext):
 
 def generate_dockstore_yaml(directory: str, publish: bool = True) -> str:
     workflows = []
-    for workflow_path in find_workflow_descriptions(directory):
+    all_workflow_paths = list(find_workflow_descriptions(directory))
+    for workflow_path in all_workflow_paths:
         test_parameter_path = f"{workflow_path.rsplit('.', 1)[0]}-tests.yml"
         workflow_entry: Dict[str, Any] = {
             # TODO: support CWL
+            "name": "main" if len(all_workflow_paths) == 1 else os.path.basename(workflow_path).replace(".ga$", ""),
             "subclass": "Galaxy",
             "publish": publish,
-            "name": "main",
             "primaryDescriptorPath": f"/{os.path.relpath(workflow_path, directory)}",
         }
         if os.path.exists(test_parameter_path):
             workflow_entry["testParameterFiles"] = [f"/{os.path.relpath(test_parameter_path, directory)}"]
+
+        with open(workflow_path) as f:
+            workflow_dict = ordered_load(f)
+
+        # add author
+        if len(workflow_dict.get("creator", [])) > 0:
+            creators = workflow_dict["creator"]
+            authors = []
+            for creator in creators:
+                author = {}
+                # Put name first
+                if "name" in creator:
+                    author["name"] = creator["name"]
+                for field, value in creator.items():
+                    if field in ["class", "name"]:
+                        continue
+                    if field == "identifier":
+                        # Check if it is an orcid:
+                        orcid = re.findall(
```

Comment:
There's also a more compact regex here: https://gist.github.com/asencis/644f174855899b873131c2cabcebeb87?permalink_comment_id=4210539#gistcomment-4210539

Should we check that the orcid is valid ?

`requests.get("https://orcid.org/0000-0002-1964-4960", headers={"Accept": "application/xml"}).status_code == 200` would tell us that.

Reply:
Great. I put it.